packages/core/src/tools/memoryTool.test.ts

```diff
@@ -34,7 +34,14 @@ vi.mock('fs', () => ({
   mkdirSync: vi.fn(),
 }));
 
-vi.mock('os');
+vi.mock('os', async (importOriginal) => {
+  const actual = await importOriginal<typeof import('os')>();
+  return {
+    ...actual,
+    default: actual,
+    homedir: vi.fn(() => path.join('/mock', 'home')),
```

Comment:
instead of doing this, can you capture the actual value and then use that to compare in the test? see below